docs/async/async-csharp-vb.rst

```diff
@@ -0,0 +1,202 @@
+Async Programming in C#/VB
+==========================
+By `Phillip Carter`_
+
+From a developer's perspective, the C#/VB async model is some nice syntax that hides complexity and allows you to focus on writing methods without worrying about things like thread pools and callbacks.
+
+The core of this are the ``async`` and ``await`` keywords (``Async`` and ``Await`` in VB.NET), which you sprinkle over otherwise normal methods. 
+
+A few important things to know before continuing:
+
+* ``Task<T>`` and ``Task`` are used to represent the return types for async code.  These are abstractions that can be thought of as "something that needs to get done and could be long-running".  `More on Task and Task<T>`_.
+* Marking a method as ``async`` tells the compiler that you're expecting to ``await`` some blocking task.
```

Comment:
I think "blocking" is the wrong term. By virtue of the code being awaitable, it's not blocking. Recall that you don't have to await the code at that point.